HW1/P4/P4_bfs.py

```diff
@@ -0,0 +1,31 @@
+import time
+
+num_of_partitions = 2
+
+def copartitioned(RDD1, RDD2):
+    "check if two RDDs are copartitioned"
+    return RDD1.partitioner == RDD2.partitioner
+
+def SSBFS( source, graph, accum ): 
+    graph = graph.map( lambda ( x, y ): ( x, ( -1, y ) ) if x != source else ( x, ( 0, y ) ), preservesPartitioning = True )
+    graph.cache()
+    i = 0
+    def adder( (x, y) ):
+        accum.add(1)
+        return (x, y[1][0])
+    while( True ):
```

Comment:
while(True) loops with break statements are often tempting, but it's generally good practice to duplicate the break condition inside of the while loop as an extra guard against infinite looping